packages/http-client/src/ceramic-http-client.ts

```diff
@@ -127,6 +127,9 @@ export default class CeramicClient implements CeramicApi {
     }
   }
 
+  /**
+   * @deprecated
+   */
   async createDocument<T extends Doctype>(doctype: string, params: DocParams, opts?: DocOpts): Promise<T> {
```

Comment:
I tried to remove this method entirely, but when I do I get the following compilation error:
```
src/ceramic-http-client.ts(83,57): error TS2345: Argument of type 'this' is not assignable to parameter of type 'Ceramic'.
  Property 'createDocument' is missing in type 'CeramicClient' but required in type 'Ceramic'.
```

I'm not actually sure what is happening here TBH.  I feel like it might be due to the circular dependency between the http-client and the 3id-did-resolver, but then shouldn't the error message say that `createDocument` is required in type `CeramicApi`?  Not in type `Ceramic`?  I'm not sure how the `Ceramic` type is in play here at all...

Reply:
Hm, yeah that's odd.

Reply:
There really isn't much else interesting besides what I posted.  Here's the error message with more of the context on either side:

```
lerna ERR! npm run build exited 2 in '@ceramicnetwork/http-client'
lerna ERR! npm run build stdout:

> @ceramicnetwork/http-client@0.11.0-rc.1 prebuild /home/spencer/js-ceramic2/packages/http-client
> npm run clean


> @ceramicnetwork/http-client@0.11.0-rc.1 clean /home/spencer/js-ceramic2/packages/http-client
> rm -rf ./lib


> @ceramicnetwork/http-client@0.11.0-rc.1 build /home/spencer/js-ceramic2/packages/http-client
> ../../node_modules/.bin/tsc -p tsconfig.json

src/ceramic-http-client.ts(83,57): error TS2345: Argument of type 'this' is not assignable to parameter of type 'Ceramic'.
  Property 'createDocument' is missing in type 'CeramicClient' but required in type 'Ceramic'.

lerna ERR! npm run build stderr:
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! @ceramicnetwork/http-client@0.11.0-rc.1 build: `../../node_modules/.bin/tsc -p tsconfig.json`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the @ceramicnetwork/http-client@0.11.0-rc.1 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/spencer/.npm/_logs/2021-04-02T21_51_08_181Z-debug.log

```